### PR TITLE
[SPARK-11079] Post-hoc review Netty-based RPC - round 1

### PR DESCRIPTION
**Pull request by rxin**
I'm going through the implementation right now for post-doc review. Adding more comments and renaming things as I go through them.

I also want to write higher level documentation about how the whole thing works -- but those will come in other pull requests.

### CODE REVIEW

**Comment by rxin** [comment]
cc @zsxwing

File: core/src/main/scala/org/apache/spark/rpc/RpcTimeout.scala [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+
+/**
+ * Associates a timeout with a description so that a when a TimeoutException occurs, additional
+ * context about the timeout can be amended to the exception message.
+ *
+ * @param duration timeout duration in seconds
+ * @param timeoutProp the configuration property that controls this timeout
+ */
+private[spark] class RpcTimeout(val duration: FiniteDuration, val timeoutProp: String)
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/rpc/RpcTimeout.scala]: I moved this into its own file -- but eventually it probably makes more sense to just have this in spark.util package for more general timeouts.

File: core/src/main/scala/org/apache/spark/rpc/RpcAddress.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+package org.apache.spark.rpc
+
+import org.apache.spark.util.Utils
+
+
+/**
+ * Address for an RPC environment, with hostname and port.
+ */
+private[spark] case class RpcAddress(host: String, port: Int) {
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/rpc/RpcAddress.scala]: this was also moved out

File: core/src/main/scala/org/apache/spark/rpc/RpcAddress.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  def hostPort: String = host + ":" + port
+
+  /** Returns a string in the form of "spark://host:port". */
+  def toSparkURL: String = "spark://" + hostPort
+
+  override def toString: String = hostPort
+}
+
+
+private[spark] object RpcAddress {
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/rpc/RpcAddress.scala]: i removed one method that was unused.

**Comment by zsxwing** [comment]
LGTM

**Comment by rxin** [comment]
I've merged this.

**Pull request closed without merging**